src/overlays/actors/ovl_En_Okuta/z_en_okuta.c

```diff
@@ -40,153 +54,1086 @@ ActorInit En_Okuta_InitVars = {
+            this->numConsecutiveProjectiles = 1;
+        }
+
+        thisx->floorHeight =
+            BgCheck_EntityRaycastFloor5(&play->colCtx, &thisx->floorPoly, &bgId, thisx, &thisx->world.pos);
+
+        if (!WaterBox_GetSurface1_2(play, &play->colCtx, thisx->world.pos.x, thisx->world.pos.z, &waterSurface,
+                                    &waterBox) ||
+            waterSurface <= thisx->floorHeight) {
+            Actor_Kill(thisx);
+        } else {
+            thisx->home.pos.y = waterSurface;
+        }
+
+        if (EN_OKUTA_GET_TYPE(thisx) == EN_OKUTA_TYPE_BLUE_OCTOROK) {
+            this->collider.base.colType = COLTYPE_HARD;
+            this->collider.base.acFlags |= AC_HARD;
+        }
+
+        thisx->targetMode = TARGET_MODE_5;
+        EnOkuta_SetupWaitToAppear(this);
+    } else {
+        ActorShape_Init(&thisx->shape, 1100.0f, ActorShadow_DrawCircle, 18.0f);
+        thisx->flags &= ~ACTOR_FLAG_TARGETABLE;
+        thisx->flags |= ACTOR_FLAG_10;
+        Collider_InitAndSetCylinder(play, &this->collider, thisx, &sProjectileCylinderInit);
+        Actor_ChangeCategory(play, &play->actorCtx, thisx, ACTORCAT_PROP);
+        this->timer = 22;
+        thisx->shape.rot.y = 0;
+        thisx->world.rot.x = -thisx->shape.rot.x;
+        thisx->shape.rot.x = 0;
+        this->actionFunc = EnOkuta_Projectile_Fly;
+        thisx->update = EnOkuta_Projectile_Update;
+        thisx->speed = 10.0f;
+    }
+}
+
+void EnOkuta_Destroy(Actor* thisx, PlayState* play) {
+    EnOkuta* this = THIS;
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+/**
+ * Covers the Octorok's body parts with ice and turns the Octorok red. Note that this is different from being frozen
+ * inside a block of ice; see `EnOkuta_FrozenInIceBlock` for how that is handled.
+ */
+void EnOkuta_Freeze(EnOkuta* this) {
+    this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX;
+    this->drawDmgEffScale = 0.6f;
+    this->drawDmgEffFrozenSteamScale = 9.0f * 0.1f;
+    this->drawDmgEffAlpha = 1.0f;
+    this->timer = 80;
+    this->collider.base.colType = COLTYPE_HIT3;
+    Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_RED, 255, COLORFILTER_BUFFLAG_OPA, 80);
+}
+
+/**
+ * Spawns ice shards on all of the Octorok's body parts that fly off in random directions. Note that this only occurs
+ * when the Octorok is frozen *without* encasing it in a block of ice.
+ */
+void EnOkuta_Thaw(EnOkuta* this, PlayState* play) {
+    if (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+        this->drawDmgEffAlpha = 0.0f;
+        Actor_SpawnIceEffects(play, &this->actor, this->bodyPartsPos, EN_OKUTA_BODYPART_MAX, 2, 0.3f, 0.2f);
+        this->collider.base.colType = COLTYPE_HIT0;
+    }
+}
+
+void EnOkuta_SpawnBubbles(EnOkuta* this, PlayState* play) {
+    s32 i;
+
+    for (i = 0; i != 10; i++) {
```

Comment:
Huh, this has to be `!=`  instead of `<`?

Reply:
No, it's just something I missed. I fixed it up.